wallet/notifications.go

```diff
@@ -358,8 +380,32 @@ type TransactionSummary struct {
 	MyOutputs   []TransactionSummaryOutput
 	Fee         dcrutil.Amount
 	Timestamp   int64
+	Type        TransactionType
 }
 
+// TransactionType decribes the which type of transaction is has been observed to be.
+// For instance, if it has a ticket as an input and a stake base reward as an ouput,
+// it is known to be a vote.
+type TransactionType int8
+
+const (
+	// TransactionTypeRegular transaction type for all non stake transactions.
+	TransactionTypeRegular TransactionType = 0
```

Comment:
Nitpicking now but this could be using iota

```
const (
    TransactionTypeRegular TransactionType = iota
    TransactionTypeTicketPurchase
    TransactionTypeVote
    TransactionTypeRevocation 
)
```